### ISSUE TITLE
Add `DockerRunContainer` task.

### ISSUE THREAD

**Issue by cdancy** [bmuschko/gradle-docker-plugin#801]
I do something similar in another project and I've seen more than a few folks ask for this (albeit more because they don't know how things work). This would be an "extra" task which basically has a dsl to configure `DockerStartContainer` and `DockerCreateContainer` and looks like:
```
task run(type: DockerRunContainer) {
    create {
        // all your favorite create args
    }
    start {
        // all your favorite start args
    }
}
```
Additionally, and internally, calls are made both to `DockerListImages` and `DockerPullImage` the former to check what images are available and the later incase we need to pull the image. This works like users expect and I've found that it's easier, at least here, for folks to get up and running quicker with this task.

WDYT?

**Comment by bmuschko** [comment]
I see a problem with maintenance here but I guess it somewhat depends on your implementation approach. Whenever one of the task adds, removes or modifies a property then it needs to be changed in the composite task as well. It will also require more tests. If it were up to me I'd say "implement that in a completely separate plugin".

**Comment by cdancy** [comment]
@bmuschko it is implemented in a separate plugin already but is definitely "controlled" and not allowed to be fudged around with so to speak to the degree that it might be here. We can hide the impl well enough behind the scenes but making it public for our end-users here is what I'm wondering about. I suspect folks might like it but am more worried/concerned about how they will use it and what type of gradle'esque issues could arise in the wild-west.

**Comment by bmuschko** [comment]
@cdancy I suspect that the "magic" that hides the implementation details will confuse users. If you decide to move forward, then the documentation for it would have to be exquisite and needs to provide examples.

**Comment by cdancy** [comment]
@bmuschko the magic I'm referring to is just our plugin which wraps this impl and provides yet another layer of DSL to make it stupid-simple for folks. The DSL itself is literally the above that I posted with us simply invoking the relevant executable method of the task. I'll mull this over some. Just wanted to throw the idea out there that it is possible, and folks have done it, but how consumable it would be to our users and what gradle users would expect in the wild might be something else.

**Comment by gkgupta** [comment]
I was looking for exactly the same thing and landed up here. Such syntactic sugar does simplify some use cases.
DSL itself is a way of hiding complexity.

**Comment by cdancy** [comment]
@gkgupta I'm currently extracting this code out into a re-usable task for the [gradle-docker-applications-plugin](https://github.com/project-aries/gradle-docker-applications-plugin) that I write. We've got an internal version here, but upon further inspection, I don't think it would be great for the broader community to consume as-is unless I went to answer questions/issues for the next 10 years. Once I extract it out, and feel comfortable with the impl, I'll reply here and see if we can move this forward again.
